### PR TITLE
Feature/update-AtomsBase-0.5

### LINKED ISSUE #89: Wrong periodic boundary condition handling
https://github.com/wexlergroup/FreeBird.jl/blob/357617fa8a2ec299f3c0b042a25d7c86d94057ee/test/test-MonteCarloMoves.jl#L335-L340

I am having a hard time figuring out why reflecting `[12.0, -2.0, 15.0]` would give `[2.0, 8.0, 5.0]`. Shouldn't it be `[8.0, 2.0, 5.0]` instead?

### PR DESCRIPTION
**Pull request by yangmr04**
# Update to use AtomsBase 0.5

This pull request includes significant changes to the `AbstractWalkers` and `EnergyEval` modules, focusing on updating functions to use new methods for handling atomic systems and periodic boundary conditions. The changes also include improvements to the handling of atomic symbols and positions within various helper functions. 

### Main changes in AtomsBase:

- `Periodic()` and `DirichletZero()` boundary conditions are deprecated. Simple booleans are used now.
- `atomic_symbol()` and other similar functions now need the atom index in the particles list to work.
- `atomic_system()` is now the easiest way to construct a flexible system.
- `boundary_conditions` and `bounding_box` are no longer fields of a system, they are now `periodicity` and `cell_vectors`, respectively.

Now using up-to-date `ExtXYZ` and `AtomsIO` packages as well, as they are changed to accommodate `AtomsBase` 0.5.

No breaking changes in the exported method, although some output now looks different.

Resolves #89.

#### AbstractWalkers Module:
* [`src/AbstractWalkers/AbstractWalkers.jl`](diffhunk://#diff-17213fcd2390efae9c289370fa59a8187b40c923ea12700042f261181acce90cR19-R22): Added exports for new helper functions `sort_components_by_atomic_number`, `split_components`, `split_components_by_chemical_species`, and `check_num_components`.
* [`src/AbstractWalkers/atomistic_walkers.jl`](diffhunk://#diff-9378974453f24cb96361f9124118caade6b0461ae1337658f0d700bf947e699dL90-R90): Updated `AtomWalker` to use list comprehensions for extracting atomic symbols and positions, improving the handling of frozen species. [[1]](diffhunk://#diff-9378974453f24cb96361f9124118caade6b0461ae1337658f0d700bf947e699dL90-R90) [[2]](diffhunk://#diff-9378974453f24cb96361f9124118caade6b0461ae1337658f0d700bf947e699dL99-R99)
* [`src/AbstractWalkers/helpers.jl`](diffhunk://#diff-5e21fa0f6714a44670e7df7798c68da2afa01224c1b71915ee05beb8fa434fdaL21-R28): Refactored `split_components` and `split_components_by_chemical_species` to use new methods for creating `FastSystem` objects. [[1]](diffhunk://#diff-5e21fa0f6714a44670e7df7798c68da2afa01224c1b71915ee05beb8fa434fdaL21-R28) [[2]](diffhunk://#diff-5e21fa0f6714a44670e7df7798c68da2afa01224c1b71915ee05beb8fa434fdaL70-R88)
* [`src/AbstractWalkers/helpers.jl`](diffhunk://#diff-5e21fa0f6714a44670e7df7798c68da2afa01224c1b71915ee05beb8fa434fdaL145-R160): Updated `sort_components_by_atomic_number` to use new methods for handling atomic symbols and positions, ensuring proper sorting and merging of species. [[1]](diffhunk://#diff-5e21fa0f6714a44670e7df7798c68da2afa01224c1b71915ee05beb8fa434fdaL145-R160) [[2]](diffhunk://#diff-5e21fa0f6714a44670e7df7798c68da2afa01224c1b71915ee05beb8fa434fdaL167-R188)

#### EnergyEval Module:
* [`src/EnergyEval/EnergyEval.jl`](diffhunk://#diff-1b2f880633e345a163879da8546cff638f1d5d0cfacc7642fdb6616a6e6601e7L17-L25): Removed exports for `sort_components_by_atomic_number` and `split_components`, as these functions are now handled within the `AbstractWalkers` module.
* [`src/EnergyEval/atomistic_energies.jl`](diffhunk://#diff-258bdf1380171fca74f6d7a745a7aecd3a1ac0098c071529aa54b894fd988bd6L18-R24): Refactored `pbc_dist` to use `periodicity` and `cell_vectors` methods for boundary conditions and box dimensions.

#### FreeBirdIO Module:
* [`src/FreeBirdIO/FreeBirdIO.jl`](diffhunk://#diff-c2c96dc5126872a2e8173d6fcab22e931354a2d5eed0afee44002b765b0d13abL39-R40): Updated `set_pbc` to use a simplified method for setting periodic boundary conditions.
* [`src/FreeBirdIO/FreeBirdIO.jl`](diffhunk://#diff-c2c96dc5126872a2e8173d6fcab22e931354a2d5eed0afee44002b765b0d13abL239-R228): Refactored `append_system` and `extract_free_par` to use new methods for handling atomic systems and periodicity. [[1]](diffhunk://#diff-c2c96dc5126872a2e8173d6fcab22e931354a2d5eed0afee44002b765b0d13abL239-R228) [[2]](diffhunk://#diff-c2c96dc5126872a2e8173d6fcab22e931354a2d5eed0afee44002b765b0d13abL326-R321)
* [`src/FreeBirdIO/FreeBirdIO.jl`](diffhunk://#diff-c2c96dc5126872a2e8173d6fcab22e931354a2d5eed0afee44002b765b0d13abL355-R344): Updated `generate_random_starting_config` and `generate_multi_type_random_starting_config` to use new methods for setting boundary conditions and creating atomic systems. [[1]](diffhunk://#diff-c2c96dc5126872a2e8173d6fcab22e931354a2d5eed0afee44002b765b0d13abL355-R344) [[2]](diffhunk://#diff-c2c96dc5126872a2e8173d6fcab22e931354a2d5eed0afee44002b765b0d13abL382-R378)

#### Tests:
* [`test/test-AbstractWalkers.jl`](diffhunk://#diff-ff0f1efa0a27f29a7003d1044d09326489b8decc078ba556764a2c202e3819beL1041-R1042): Updated tests to verify the new methods for handling cell vectors and periodicity in atomic systems. [[1]](diffhunk://#diff-ff0f1efa0a27f29a7003d1044d09326489b8decc078ba556764a2c202e3819beL1041-R1042) [[2]](diffhunk://#diff-ff0f1efa0a27f29a7003d1044d09326489b8decc078ba556764a2c202e3819beL1058-R1059) [[3]](diffhunk://#diff-ff0f1efa0a27f29a7003d1044d09326489b8decc078ba556764a2c202e3819beL1082-R1083)
* [`test/test-EnergyEval.jl`](diffhunk://#diff-8d3c3134897cf2d73b979ddd655d09877e9395db2f7d19760d6b657115a7ecc1L69-R70): Updated tests for `pbc_dist` to use the new methods for setting non-periodic boundary conditions.

Documentation preview: https://wexlergroup.github.io/FreeBird.jl/previews/PR90/

### CODE REVIEW

File: src/FreeBirdIO/FreeBirdIO.jl [hunk 1]
```diff
@@ -36,19 +36,8 @@ Set the periodic boundary conditions for a system of atoms.
...
-        elseif pbc[i] == true
-            push!(pbc_conditions, Periodic())
-        else
-            error("Unsupported boundary condition: $(pbc[i])")
-        end
-    end
-    pbc_conditions = Vector{BoundaryCondition}(pbc_conditions)
-    return FlexibleSystem(at;boundary_conditions=pbc_conditions)
+function set_pbc(at::Atoms, pbc::Vector{Bool})
+    return FlexibleSystem(at, periodicity=Tuple(pbc))
```
Comment (junchichen21) [inline src/FreeBirdIO/FreeBirdIO.jl:40]: Cool, the updated function is straighter!

Comment (yangmr04) [reply src/FreeBirdIO/FreeBirdIO.jl:40]: It's because both `DirichletZero()` and `Periodic()` are no longer in use in AtomsBase 0.5. Now it has to be done this way.

**Review by junchichen21 — APPROVED** [review]
I saw many changes both in the source codes and test codes due to employing the newer version of package `AtomsBase 0.5`. From my perspectives, these changes make the code more readable. Good work!
